pkg/eval/vals/aliased_types.go

```diff
@@ -7,8 +7,37 @@ import (
 	"github.com/xiaq/persistent/vector"
 )
 
-// File is an alias for *os.File.
-type File = *os.File
+// File wraps *os.File so that Elvish can treat the object as a read-only psuedo-map.
+type File struct {
+	OsFile *os.File `elvish:"hidden"`
+	// IoMode causes `$f[io-mode]` to return "ro" if the file is read-only, "rw" if it is
+	// read-write, and "wo" if it is write-only.
+	IoMode string
+}
+
+// IsStructMap tags File as a psuedo-map.
+func (File) IsStructMap() {}
+
+// IsClosed causes `$f[is-closed]` to return true if the file object is closed.
+func (f File) IsClosed() bool {
+	fdPtr := f.OsFile.Fd()
+	return fdPtr == ^uintptr(0)
+}
+
+// Fd causes `$f[fd]` to return the numeric file descriptor of the embedded os.File object.
+//
+// We coerce it to a float64 since that is currently the only Elvish numeric type and we don't want
+// the value coerced to a string.
+func (f File) Fd() float64 {
+	fdPtr := f.OsFile.Fd()
+	if fdPtr == ^uintptr(0) {
+		return -1
+	}
+	return float64(fdPtr)
```

Comment:
The default for exporting integer values to Elvish is to convert it to a string (https://github.com/elves/elvish/blob/f5aaba06911ac9d47d40adcb1f2fe615c77c741e/pkg/eval/vals/conversion.go#L110). Before Elvish gets an integer type, better to keep the consistency.

Reply:
Done.